### PR TITLE
Populate inchi dict batch1

### PR DESCRIPTION
**Pull request by sumitnegi7**
How to verify the data =>

Search compound name or CAS no in `PubChem` and match the values. More info in issue #7

### CODE REVIEW

**Comment by sumitnegi7** [comment]
Sir @maneetgoyal  have added `CAS` as an optional property. It can be made mandatory too if we fill it up for all existing compounds. Is optional okay for now?

**Comment by maneetgoyal** [comment]
> Sir @maneetgoyal have added `CAS` as an optional property. It can be made mandatory too if we fill it up for all existing compounds. Is optional okay for now?

Try to add it for all the compounds & make it mandatory.

**Review by qur786 — CHANGES_REQUESTED** [review]

**Comment by saikat-samanta** [comment]
looks good 👍

**Comment by sumitnegi7** [comment]
Sir @maneetgoyal  please check `CAS` has been made mandatory and added or all compounds.

File: src/data/inchi-dictionary.ts [hunk 1]
```diff
@@ -2,430 +2,698 @@ import { INCHIProps } from '../interfaces/utils';
 
 export const INCHIDictionary: Record<string, INCHIProps> = {
   Aluminum: {
+    CAS: '7429-92-5',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:5]:
```suggestion
    CAS: '7429-90-5',
```

File: src/data/inchi-dictionary.ts [hunk 2]
```diff
@@ -2,430 +2,698 @@ import { INCHIProps } from '../interfaces/utils';
...
+
   NormalAluminumAcetate: {
+    CAS: '139-12-8',
     inchiString: 'InChI=1S/3C2H4O2.Al/c3*1-2(3)4;/h3*1H3,(H,3,4);/q;;;+3/p-3',
     inchiKey: 'WCOATMADISNSBV-UHFFFAOYSA-K',
     molecularWeight: 204.11,
   },
+
   BasicAluminumAcetate: {
+    CAS: '142-03-0',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:19]:
```suggestion
    CAS: '',
```

File: src/data/inchi-dictionary.ts [hunk 3]
```diff
@@ -2,430 +2,698 @@ import { INCHIProps } from '../interfaces/utils';
...
     inchiString:
       'InChI=1S/2C2H4O2.Al.H2O/c2*1-2(3)4;;/h2*1H3,(H,3,4);;1H2/q;;+3;/p-3',
     inchiKey: 'HQQUTGFAWJNQIP-UHFFFAOYSA-K',
     molecularWeight: 162.08,
   },
+
   AluminumBromide: {
+    CAS: '',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:27]:
```suggestion
    CAS: '7727-15-3',
```

File: src/data/inchi-dictionary.ts [hunk 4]
```diff
@@ -2,430 +2,698 @@ import { INCHIProps } from '../interfaces/utils';
...
+
   AluminumChloride: {
+    CAS: '7446-70-0',
     inchiString: 'InChI=1S/Al.3ClH/h;3*1H/q+3;;;/p-3',
     inchiKey: 'VSCWAEJMTAWNJL-UHFFFAOYSA-K',
     molecularWeight: 133.34,
   },
+
   AluminumChlorideHexahydrate: {
+    CAS: '',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:55]:
```suggestion
    CAS: '7784-13-6',
```

File: src/data/inchi-dictionary.ts [hunk 5]
```diff
@@ -2,430 +2,698 @@ import { INCHIProps } from '../interfaces/utils';
...
     inchiString: 'InChI=1S/Al.3ClH.6H2O/h;3*1H;6*1H2/q+3;;;;;;;;;/p-3',
     inchiKey: 'JGDITNMASUZKPW-UHFFFAOYSA-K',
     molecularWeight: 241.43,
   },
+
   AluminumFluorideHydrate: {
+    CAS: '15098-87-0',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:62]:
```suggestion
    CAS: '32287-65-3',
```

File: src/data/inchi-dictionary.ts [hunk 6]
```diff
@@ -2,430 +2,698 @@ import { INCHIProps } from '../interfaces/utils';
...
+
   DialuminumHeptafluoride: {
+    CAS: '',
     inchiString: '',
     inchiKey: '',
     molecularWeight: 294.06,
   },
+
   AluminumHydroxide: {
+    CAS: 'InChI=1S/Al.3H2O/h;3*1H2/q+3;;;/p-3',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:76]:
```suggestion
    CAS: '',
```

File: src/data/inchi-dictionary.ts [hunk 7]
```diff
@@ -2,430 +2,698 @@ import { INCHIProps } from '../interfaces/utils';
...
+
   AluminumOxide: {
+    CAS: '1344-28-1',
     inchiString: 'InChI=1S/2Al.3O/q2*+3;3*-2',
     inchiKey: 'PNEYBMLMFCGWSK-UHFFFAOYSA-N',
     molecularWeight: 101.96,
   },
+
   AluminumOxideCorundum: {
+    CAS: '',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:105]:
```suggestion
    CAS: '1302-74-5',
```

File: src/data/inchi-dictionary.ts [hunk 8]
```diff
@@ -2,430 +2,698 @@ import { INCHIProps } from '../interfaces/utils';
...
+
   AmmoniumAuricyanide: {
+    CAS: '',
     inchiString: '',
     inchiKey: '',
     molecularWeight: 337.09,
   },
+
   AmmoniumBicarbonate: {
+    CAS: '10361-29-2',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:225]:
```suggestion
    CAS: '1066-33-7',
```

File: src/data/inchi-dictionary.ts [hunk 9]
```diff
@@ -2,430 +2,698 @@ import { INCHIProps } from '../interfaces/utils';
...
+
   AmmoniumCarbonate: {
+    CAS: '',
     inchiString: 'InChI=1S/CH2O3.2H3N.H2O/c2-1(3)4;;;/h(H2,2,3,4);2*1H3;1H2',
     inchiKey: 'MMVILKIFGQPOLE-UHFFFAOYSA-N',
     molecularWeight: 114.1,
   },
+
   AmmoniumCarbonateCarbamate: {
+    CAS: '',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:246]:
```suggestion
    CAS: '8000-73-5',
```

File: src/data/inchi-dictionary.ts [hunk 10]
```diff
@@ -2,430 +2,698 @@ import { INCHIProps } from '../interfaces/utils';
...
+
   AmmoniumChlorideSalammoniac: {
+    CAS: '12125-02-9',
     inchiString: 'InChI=1S/ClH.H3N/h1H;1H3',
     inchiKey: 'NLXLAEXVIDQMFP-UHFFFAOYSA-N',
     molecularWeight: 53.49,
   },
+
   AmmoniumChloroplatinate: {
+    CAS: '',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:268]:
```suggestion
    CAS: '16919-58-7',
```

File: src/data/inchi-dictionary.ts [hunk 11]
```diff
@@ -2,430 +2,698 @@ import { INCHIProps } from '../interfaces/utils';
...
     inchiString: 'InChI=1S/6ClH.2H3N.Pt/h6*1H;2*1H3;/q;;;;;;;;+4/p-4',
     inchiKey: 'PCCGQTHFYHJATL-UHFFFAOYSA-J',
     molecularWeight: 443.87,
   },
+
   AmmoniumChloroplatinite: {
+    CAS: '',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:275]:
```suggestion
    CAS: '13820-41-2',
```

File: src/data/inchi-dictionary.ts [hunk 12]
```diff
@@ -2,430 +2,698 @@ import { INCHIProps } from '../interfaces/utils';
...
+
   AmmoniumChlorostannate: {
+    CAS: '',
     inchiString: 'InChI=1S/6ClH.2H3N.Sn/h6*1H;2*1H3;/q;;;;;;;;+4/p-4',
     inchiKey: 'UDMAVAHFQCVSSE-UHFFFAOYSA-J',
     molecularWeight: 367.5,
   },
+
   AmmoniumChromate: {
+    CAS: '7788-98-9',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:289]:
```suggestion
    CAS: '',
```

File: src/data/inchi-dictionary.ts [hunk 13]
```diff
@@ -2,430 +2,698 @@ import { INCHIProps } from '../interfaces/utils';
...
+
   AmmoniumFluoride: {
+    CAS: '12125-01-8',
     inchiString: 'InChI=1S/FH.H3N/h1H;1H3',
     inchiKey: 'LDDQLRUQCUTJBB-UHFFFAOYSA-N',
     molecularWeight: 37.04,
   },
+
   AmmoniumFluorideAcid: {
+    CAS: '',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:324]:
```suggestion
    CAS: '1341-49-7',
```

**Review by maneetgoyal — CHANGES_REQUESTED** [review]
Loads of issues. Haven't reviewed all.

File: src/data/inchi-dictionary.ts [hunk 14]
```diff
@@ -2,430 +2,698 @@ import { INCHIProps } from '../interfaces/utils';
...
+
   NormalAluminumAcetate: {
+    CAS: '139-12-8',
     inchiString: 'InChI=1S/3C2H4O2.Al/c3*1-2(3)4;/h3*1H3,(H,3,4);/q;;;+3/p-3',
     inchiKey: 'WCOATMADISNSBV-UHFFFAOYSA-K',
     molecularWeight: 204.11,
   },
+
   BasicAluminumAcetate: {
+    CAS: '',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:19]:
```suggestion
    CAS: '142-03-0',
```

**Comment by sumitnegi7** [comment]
@qur786  please try to review this  when you are free.

File: src/data/inchi-dictionary.ts [hunk 15]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
     inchiKey: '',
     molecularWeight: 524.44,
   },
+
   AluminumSulfate: {
+    CAS: '10043-01-3',
     inchiString:
       'InChI=1S/2Al.3H2O4S/c;;3*1-5(2,3)4/h;;3*(H2,1,2,3,4)/q2*+3;;;/p-6',
     inchiKey: 'DIZPMCHEQGEION-UHFFFAOYSA-H',
     molecularWeight: 342.15,
```
Comment (qur786) [inline src/data/inchi-dictionary.ts:158]:
```suggestion
    molecularWeight: 342.2,
```

File: src/data/inchi-dictionary.ts [hunk 16]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
     inchiString: '',
     inchiKey: '',
     molecularWeight: 916.56,
   },
+
   Ammonia: {
+    CAS: '7664-41-7',
     inchiString: 'InChI=1S/H3N/h1H3',
     inchiKey: 'QGZKDVFQNNGYKY-UHFFFAOYSA-N',
     molecularWeight: 17.03,
```
Comment (qur786) [inline src/data/inchi-dictionary.ts:207]:
```suggestion
    molecularWeight: 17.031,
```

File: src/data/inchi-dictionary.ts [hunk 17]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
     inchiString: '',
     inchiKey: '',
     molecularWeight: 337.09,
   },
+
   AmmoniumBicarbonate: {
+    CAS: '1066-33-7',
     inchiString: 'InChI=1S/CH2O3.H3N/c2-1(3)4;/h(H2,2,3,4);1H3',
     inchiKey: 'ATRRKUHOCOJYRX-UHFFFAOYSA-N',
     molecularWeight: 79.06,
```
Comment (qur786) [inline src/data/inchi-dictionary.ts:228]:
```suggestion
    molecularWeight: 79.058,
```

File: src/data/inchi-dictionary.ts [hunk 18]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
+    CAS: '12124-97-9',
     inchiString: 'InChI=1S/BrH.H3N/h1H;1H3',
     inchiKey: 'SWLVFNYSXGMGBS-UHFFFAOYSA-N',
     molecularWeight: 97.94,
   },
+
   AmmoniumCarbonate: {
+    CAS: '',
     inchiString: 'InChI=1S/CH2O3.2H3N.H2O/c2-1(3)4;;;/h(H2,2,3,4);2*1H3;1H2',
     inchiKey: 'MMVILKIFGQPOLE-UHFFFAOYSA-N',
```
Comment (qur786) [inline src/data/inchi-dictionary.ts:241]:
```suggestion
    inchiKey: 'PRKQVKDSMLBJBJ-UHFFFAOYSA-N',
```

File: src/data/inchi-dictionary.ts [hunk 19]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
+
   AmmoniumBromide: {
+    CAS: '12124-97-9',
     inchiString: 'InChI=1S/BrH.H3N/h1H;1H3',
     inchiKey: 'SWLVFNYSXGMGBS-UHFFFAOYSA-N',
     molecularWeight: 97.94,
   },
+
   AmmoniumCarbonate: {
+    CAS: '',
```
Comment (qur786) [inline src/data/inchi-dictionary.ts:239]:
```suggestion
    CAS: '506-87-6',
```

File: src/data/inchi-dictionary.ts [hunk 20]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
     inchiString: 'InChI=1S/CH2O3.2H3N.H2O/c2-1(3)4;;;/h(H2,2,3,4);2*1H3;1H2',
     inchiKey: 'MMVILKIFGQPOLE-UHFFFAOYSA-N',
     molecularWeight: 114.1,
```
Comment (qur786) [inline src/data/inchi-dictionary.ts:242]:
```suggestion
    molecularWeight: 96.09,
```

**Review by qur786 — CHANGES_REQUESTED** [review]
Sumit I have only reviewed half of the data yet , so please don't merge after correcting the data as  i have to do the review for another half . I am going to attend IIoT classes.

File: src/data/inchi-dictionary.ts [hunk 21]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
 
 export const INCHIDictionary: Record<string, INCHIProps> = {
   Aluminum: {
+    CAS: '7429-90-5',
     inchiString: 'InChI=1S/Al',
     inchiKey: 'XAGFODPZIPBFFR-UHFFFAOYSA-N',
     molecularWeight: 26.98,
```
[the remark this answers is not in the document]
Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:8]: Most of the comments have suggestions for molecular weight that is present before the PR and rest values are mapped according to it only.So molecular weight cannot and should not be changed.

File: src/data/inchi-dictionary.ts [hunk 22]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
     inchiKey: 'WCOATMADISNSBV-UHFFFAOYSA-K',
     molecularWeight: 204.11,
   },
+
   BasicAluminumAcetate: {
+    CAS: '142-03-0',
     inchiString:
       'InChI=1S/2C2H4O2.Al.H2O/c2*1-2(3)4;;/h2*1H3,(H,3,4);;1H2/q;;+3;/p-3',
     inchiKey: 'HQQUTGFAWJNQIP-UHFFFAOYSA-K',
     molecularWeight: 162.08,
```
[the remark this answers is not in the document]
Comment (maneetgoyal) [reply src/data/inchi-dictionary.ts:23]: `163.08` looks correct. Perry's handbook may have wrong info.

File: src/data/inchi-dictionary.ts (hunk 21 above)
[the remark this answers is not in the document]
Comment (qur786) [reply src/data/inchi-dictionary.ts:8]: ok sumit

File: src/data/inchi-dictionary.ts [hunk 23]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
+
   AmmoniumFormate: {
+    CAS: '540-69-2',
     inchiString: 'InChI=1S/CH2O2.H3N/c2-1-3;/h1H,(H,2,3);1H3',
     inchiKey: 'VZTDIZULWFCMLS-UHFFFAOYSA-N',
     molecularWeight: 63.06,
   },
+
   AmmoniumHydrosulfide: {
+    CAS: '',
```
Comment (qur786) [inline src/data/inchi-dictionary.ts:338]: 12124-99-1 is given as CAS in PubChem but does not correspond to any compound in "Sigmaaldrich"

File: src/data/inchi-dictionary.ts [hunk 24]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
     inchiString: 'InChI=1S/H3N.H2S/h1H3;1H2',
     inchiKey: 'HIVLDXAAFGCOFU-UHFFFAOYSA-N',
     molecularWeight: 51.11,
   },
+
   AmmoniumHydroxide: {
+    CAS: '12168-30-8',
```
Comment (qur786) [inline src/data/inchi-dictionary.ts:345]:
```suggestion
    CAS: '1336-21-6',
```

File: src/data/inchi-dictionary.ts [hunk 25]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
+    CAS: '13106-76-8',
     inchiString: 'InChI=1S/Mo.2H3N.4O/h;2*1H3;;;;/q;;;;;2*-1/p+2',
     inchiKey: 'APUPEJJSWDHEBO-UHFFFAOYSA-P',
     molecularWeight: 196.01,
   },
+
   AmmoniumHeptamolybdate: {
+    CAS: '',
     inchiString:
       'InChI=1S/7Mo.6H3N.10H2O.18O/h;;;;;;;6*1H3;10*1H2;;;;;;;;;;;;;;;;;;/q;;;;3*+2;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;/p-6',
```
Comment (qur786) [inline src/data/inchi-dictionary.ts:361]: InChI=1S/7Mo.6H3N.18H2O.10O/h;;;;;;;6*1H3;18*1H2;;;;;;;;;;/q;;;2*+1;2*+2;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;2*-1/p-4

DVXZHPYMEZBKFA-UHFFFAOYSA-J

Molecular Weight = 1254.1

File: src/data/inchi-dictionary.ts [hunk 26]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
   },
+
   AmmoniumOsmochloride: {
+    CAS: '12125-08-5',
     inchiString: 'InChI=1S/6ClH.2H3N.Os/h6*1H;2*1H3;/q;;;;;;;;+4/p-4',
     inchiKey: 'SRBXXQDKBKTWOC-UHFFFAOYSA-J',
     molecularWeight: 439,
   },
+
   AmmoniumOxalate: {
```
Comment (qur786) [inline src/data/inchi-dictionary.ts:394]: Ammonium oxalate
CAS : 1113-38-8
Molecular Weight : 124.10
InChI=1S/C2H2O4.2H3N/c3-1(4)2(5)6;;/h(H,3,4)(H,5,6);2*1H3
VBIXEXWLHSRNKB-UHFFFAOYSA-N

File: src/data/inchi-dictionary.ts [hunk 27]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
+    CAS: '',
     inchiString:
       'InChI=1S/C2H2O4.2H3N.H2O/c3-1(4)2(5)6;;;/h(H,3,4)(H,5,6);2*1H3;1H2',
     inchiKey: 'MSMNVXKYCPHLLN-UHFFFAOYSA-N',
     molecularWeight: 142.11,
   },
+
   AmmoniumOxalateAcid: {
```
Comment (qur786) [inline src/data/inchi-dictionary.ts:402]: AmmoniumOxalateAcid

CAS : 6009-70-7
Molecular Weight : 6009-70-7
InChI=1S/C2H2O4.2H3N.H2O/c3-1(4)2(5)6;;;/h(H,3,4)(H,5,6);2*1H3;1H2
MSMNVXKYCPHLLN-UHFFFAOYSA-N

**Review by qur786 — CHANGES_REQUESTED** [review]

**Review by qur786 — CHANGES_REQUESTED** [review]

File: src/data/inchi-dictionary.ts [hunk 28]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
   AluminumChloride: {
+    CAS: '7446-70-0',
     inchiString: 'InChI=1S/Al.3ClH/h;3*1H/q+3;;;/p-3',
     inchiKey: 'VSCWAEJMTAWNJL-UHFFFAOYSA-K',
     molecularWeight: 133.34,
   },
+
   AluminumChlorideHexahydrate: {
+    CAS: '7784-13-6',
     inchiString: 'InChI=1S/Al.3ClH.6H2O/h;3*1H;6*1H2/q+3;;;;;;;;;/p-3',
```
[the remark this answers is not in the document]
Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:56]: That is fine

File: src/data/inchi-dictionary.ts [hunk 29]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
   },
+
   AluminumOxide: {
+    CAS: '1344-28-1',
     inchiString: 'InChI=1S/2Al.3O/q2*+3;3*-2',
     inchiKey: 'PNEYBMLMFCGWSK-UHFFFAOYSA-N',
     molecularWeight: 101.96,
   },
+
   AluminumOxideCorundum: {
```
[the remark this answers is not in the document]
Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:104]: Sir @maneetgoyal  should i make this change?

File: src/data/inchi-dictionary.ts (hunk 20 above)
Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:242]: Sir @maneetgoyal  here there is a lot of diff in perry handbook and [sigma aldrich](https://www.sigmaaldrich.com/IN/en/substance/ammoniumcarbonate9609506876?context=product)

File: src/data/inchi-dictionary.ts (hunk 23 above)
Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:338]: Molecular weight does not match so let's leave it.

File: src/data/inchi-dictionary.ts (hunk 25 above)
Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:361]: https://www.sigmaaldrich.com/IN/en/substance/ammoniumheptamolybdatetetrahydrate12358612054852?context=product

File: src/data/inchi-dictionary.ts (hunk 26 above)
Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:394]: https://www.sigmaaldrich.com/IN/en/substance/ammoniumoxalatemonohydrate142116009707?context=product

Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:394]: Its okay checked molecular formula too in perry its `(NH4)2C2O4⋅H2O`

File: src/data/inchi-dictionary.ts (hunk 27 above)
Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:402]: Sir  @maneetgoyal  molecular weight and formula does not match please have a look

File: src/data/inchi-dictionary.ts [hunk 30]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
   },
+
   AmmoniumDibasicphosphate: {
+    CAS: '',
     inchiString: 'InChI=1S/2H3N.H3O4P/c;;1-5(2,3)4/h2*1H3;(H3,1,2,3,4)',
     inchiKey: 'MNNHAPBLZZVQHP-UHFFFAOYSA-N',
     molecularWeight: 132.06,
   },
+
   AmmoniumMetaphosphate: {
```
[the remark this answers is not in the document]
Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:439]: Molecular weight does not match in perry it is 388.04 there

File: src/data/inchi-dictionary.ts [hunk 31]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
   },
+
   AmmoniumSulfide: {
+    CAS: '12135-76-1',
     inchiString: 'InChI=1S/2H3N.H2S/h2*1H3;1H2',
     inchiKey: 'UXXGWUQNRMPNKH-UHFFFAOYSA-N',
     molecularWeight: 68.14,
   },
+
   AmmoniumPentasulfide: {
```
[the remark this answers is not in the document]
Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:488]: Does not match perry hsndbook

File: src/data/inchi-dictionary.ts [hunk 32]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
     inchiString: 'InChI=1S/2H3N.H2O3S.H2O/c;;1-4(2)3;/h2*1H3;(H2,1,2,3);1H2',
     inchiKey: 'ASPXBYAQZVXSNS-UHFFFAOYSA-N',
     molecularWeight: 134.16,
   },
+
   AmmoniumSulfiteAcid: {
+    CAS: '10196-04-0',
     inchiString: 'InChI=1S/H3N.H2O3S/c;1-4(2)3/h1H3;(H2,1,2,3)',
     inchiKey: 'ZETCGWYACBNPIH-UHFFFAOYSA-N',
     molecularWeight: 99.11,
```
[the remark this answers is not in the document]
Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:506]: https://pubchem.ncbi.nlm.nih.gov/compound/56846439#section=Names-and-Identifiers  this matches the one you mentioned for AmmoniumSulfiteAcid does not match molecular formula or weight from perry handbook.

File: src/data/inchi-dictionary.ts [hunk 33]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
   },
+
   AmmoniumSulfite: {
+    CAS: '',
     inchiString: 'InChI=1S/2H3N.H2O3S.H2O/c;;1-4(2)3;/h2*1H3;(H2,1,2,3);1H2',
     inchiKey: 'ASPXBYAQZVXSNS-UHFFFAOYSA-N',
     molecularWeight: 134.16,
   },
+
   AmmoniumSulfiteAcid: {
```
[the remark this answers is not in the document]
Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:502]: Fomua in perry `NH4HSO3` formula in the link `H8N2O3S`

File: src/data/inchi-dictionary.ts [hunk 34]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
+    CAS: '10196-04-0',
     inchiString: 'InChI=1S/H3N.H2O3S/c;1-4(2)3/h1H3;(H2,1,2,3)',
     inchiKey: 'ZETCGWYACBNPIH-UHFFFAOYSA-N',
```
[the remark this answers is not in the document]
Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:505]: https://github.com/manufac-analytics/perry/pull/33#discussion_r681524036

File: src/data/inchi-dictionary.ts [hunk 35]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
   AmmoniumSulfite: {
+    CAS: '',
     inchiString: 'InChI=1S/2H3N.H2O3S.H2O/c;;1-4(2)3;/h2*1H3;(H2,1,2,3);1H2',
     inchiKey: 'ASPXBYAQZVXSNS-UHFFFAOYSA-N',
     molecularWeight: 134.16,
   },
+
   AmmoniumSulfiteAcid: {
+    CAS: '10196-04-0',
     inchiString: 'InChI=1S/H3N.H2O3S/c;1-4(2)3/h1H3;(H2,1,2,3)',
```
[the remark this answers is not in the document]
Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:504]: https://github.com/manufac-analytics/perry/pull/33#discussion_r681524036

File: src/data/inchi-dictionary.ts [hunk 36]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
+
   AmmoniumChlorostannate: {
+    CAS: '',
     inchiString: 'InChI=1S/6ClH.2H3N.Sn/h6*1H;2*1H3;/q;;;;;;;;+4/p-4',
     inchiKey: 'UDMAVAHFQCVSSE-UHFFFAOYSA-J',
     molecularWeight: 367.5,
   },
+
   AmmoniumChromate: {
+    CAS: '',
```
[the remark this answers is not in the document]
Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:289]: Searched the CAS in pubchem a lot of data is missing from the page so let's not update CAS

File: src/data/inchi-dictionary.ts (hunk 20 above)
Comment (maneetgoyal) [reply src/data/inchi-dictionary.ts:242]: The mol. form. is different in both. In Perry, it is: `(NH4)2CO3.H2O`. Sigma is `(NH4)2CO3`.

File: src/data/inchi-dictionary.ts [hunk 37]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
     inchiString: 'InChI=1S/BrH.H3N/h1H;1H3',
     inchiKey: 'SWLVFNYSXGMGBS-UHFFFAOYSA-N',
     molecularWeight: 97.94,
   },
+
   AmmoniumCarbonate: {
+    CAS: '506-87-6',
     inchiString: 'InChI=1S/CH2O3.2H3N.H2O/c2-1(3)4;;;/h(H2,2,3,4);2*1H3;1H2',
-    inchiKey: 'MMVILKIFGQPOLE-UHFFFAOYSA-N',
+    inchiKey: 'PRKQVKDSMLBJBJ-UHFFFAOYSA-N',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:241]: - Older InChI key is correct.
- Ref: https://pubchem.ncbi.nlm.nih.gov/compound/86740279

File: src/data/inchi-dictionary.ts (hunk 20 above)
Comment (maneetgoyal) [reply src/data/inchi-dictionary.ts:242]: No change needed in `molWt`.

File: src/data/inchi-dictionary.ts (hunk 36 above)
[the remark this answers is not in the document]
Comment (maneetgoyal) [reply src/data/inchi-dictionary.ts:289]: `7788-98-9` is preferred. Ref.: https://pubchem.ncbi.nlm.nih.gov/compound/24595 is good enough.

File: src/data/inchi-dictionary.ts (hunk 23 above)
Comment (maneetgoyal) [reply src/data/inchi-dictionary.ts:338]: `12124-99-1` is preferred. Ref.: https://pubchem.ncbi.nlm.nih.gov/compound/25515#section=Molecular-Formula

File: src/data/inchi-dictionary.ts (hunk 25 above)
Comment (maneetgoyal) [reply src/data/inchi-dictionary.ts:361]: Exisiting entry is good enough. No change needed

File: src/data/inchi-dictionary.ts (hunk 26 above)
Comment (maneetgoyal) [reply src/data/inchi-dictionary.ts:394]: No change needed.

File: src/data/inchi-dictionary.ts [hunk 38]
```diff
@@ -2,430 +2,825 @@ import { INCHIProps } from '../interfaces/utils';
...
   AmmoniumOxalate: {
+    CAS: '6009-70-7',
     inchiString:
       'InChI=1S/C2H2O4.2H3N.H2O/c3-1(4)2(5)6;;;/h(H,3,4)(H,5,6);2*1H3;1H2',
     inchiKey: 'MSMNVXKYCPHLLN-UHFFFAOYSA-N',
     molecularWeight: 142.11,
   },
+
   AmmoniumOxalateAcid: {
+    CAS: '6009-70-7',
```
Comment (maneetgoyal) [inline src/data/inchi-dictionary.ts:403]:
```suggestion
    CAS: '5972-73-6',
```

File: src/data/inchi-dictionary.ts (hunk 30 above)
[the remark this answers is not in the document]
Comment (maneetgoyal) [reply src/data/inchi-dictionary.ts:439]: No change needed

**Review by maneetgoyal — CHANGES_REQUESTED** [review]
Partial review so far

File: src/data/inchi-dictionary.ts (hunk 23 above)
Comment (qur786) [reply src/data/inchi-dictionary.ts:338]: @sumitnegi7 did you apply this change ?

Comment (sumitnegi7) [reply src/data/inchi-dictionary.ts:338]: Didn't work on the inchi PR yesterday @qur786  bringing the change now.

File: src/data/inchi-dictionary.ts [hunk 39]
```diff
@@ -2,422 +2,815 @@ import { INCHIProps } from "../interfaces/utils";
...
+    inchiString: "InChI=1S/C4H7N/c1-2-3-4-5/h2-3H2,1H3",
+    inchiKey: "KVNRLNFWIYMESJ-UHFFFAOYSA-N",
+    molecularWeight: 69.11
+  },
+
+  CarbonDioxide: {
+    CAS: "124-38-9",
+    inchiString: "InChI=1S/CO2/c2-1-3",
+    inchiKey: "CURLTUGMZLYLDI-UHFFFAOYSA-N",
+    molecularWeight: 44.009
```
Comment (saikat-samanta) [inline src/data/inchi-dictionary.ts:814]: Looks good 👍

**Review by qur786 — APPROVED** [review]
looks good 👍

**Comment by maneetgoyal** [comment]
Closing it

**Pull request closed without merging**